lib/galaxy/managers/workflows.py

```diff
@@ -420,14 +421,143 @@ def _workflow_to_dict_run(self, trans, stored):
+                        if param.tag not in user_permission:
+                            user_permission[param.tag] = {}
+                        if param.text is not None:
+                            for s in param.text.split(','):
+                                user_permission[param.tag][s] = {}
+            return user_permission
+
+        # Creates the data structure for the parameters that a set permissions can access.
+        def _get_param_list(param_definitions_root, user_permissions):
+            param_list = []
+            for param_elem in param_definitions_root.findall("param"):
+                attr = param_elem.attrib
+                if attr['name'] in user_permissions:
+                    # Allow 'select' type parameters to be used
+                    if attr['type'] == 'select':
+                        option_data = []
+                        reject_list = []
+                        for option_elem in param_elem.findall("option"):
+                            if option_elem.attrib['value'] in user_permissions[attr['name']]:
+                                option_data.append({
+                                    'label': option_elem.attrib['label'],
+                                    'value': option_elem.attrib['value']
+                                })
+                            else:
+                                reject_list.append(option_elem.attrib['label'])
+                        attr['data'] = option_data
+
+                        attr_help = ""
+                        if 'help' in attr:
+                            attr_help = attr['help']
+                        attr['help'] = _get_select_help_message(attr_help, reject_list)
+                    param_list.append(attr)
+            return param_list
+
+        # Creates the html text for a select message when parameters are not avaliable
+        def _get_select_help_message(help, reject_list):
+            if reject_list:
+                help += "<br/><br/>The following options are available but disabled.<br/>" + str(
+                    reject_list) + "<br/>If you believe this is a mistake, please contact your Galaxy admin."
+            return help
+
+        params = []
+        if self._workflow_options_xml_validation():
+            if os.path.exists(self.app.config.workflow_resource_params_file):
+                resource_param_file = self.app.config.workflow_resource_params_file
+                try:
+                    resource_definitions = util.parse_xml(resource_param_file)
+                except Exception as e:
+                    raise log.exception(e, resource_param_file)
+
+                permissions = _get_user_permissions(resource_definitions.getroot().find("groups"), stored.user)
+                params = _get_param_list(resource_definitions.getroot().find("parameters"), permissions)
+
+        return params
+
+    def _workflow_options_xml_validation(self):
```

Comment:
Would this function be better suited in another file?
Also, I would like to allow this to be callable by an admin to verify if their file well formed.